frontend/src/components/custom-tools/pdf-viewer/PdfViewer.jsx

```diff
@@ -12,13 +12,19 @@ try {
   RenderHighlights =
     require("../../../plugins/pdf-highlight/RenderHighlights").RenderHighlights;
 } catch (err) {
-  // Do nothing, No plugin will be loaded.
+  // Do nothing, no plugin will be loaded.
 }
 
 function PdfViewer({ fileUrl, highlightData }) {
   const newPlugin = defaultLayoutPlugin();
   const pageNavigationPluginInstance = pageNavigationPlugin();
+  const { jumpToPage } = pageNavigationPluginInstance;
   const parentRef = useRef(null);
+  function removeZerosAndDeleteIfAllZero(highlightData) {
+    return highlightData.filter(
+      (innerArray) => !innerArray.every((value) => value === 0)
+    ); // Remove arrays where all values are 0
+  }
```

Comment:
@vishnuszipstack NIT: The function can be optimized by using the `.some()` method instead of `.every()`. The `.some()` method returns true as soon as it finds an element that satisfies the condition, which can improve performance by avoiding unnecessary checks through the entire array.

```suggestion
  function removeZerosAndDeleteIfAllZero(highlightData) {
    return highlightData.filter(
      (innerArray) => innerArray.some((value) => value !== 0)
    ); // Keep arrays that contain at least one non-zero value
  }
```